opts/envdir_test.go

```diff
@@ -0,0 +1,65 @@
+package opts
+
+import (
+	"testing"
+)
+
+func TestParseEnvDir(t *testing.T) {
+	fixtures := []struct {
+		dir     string
+		results []string
+		failure bool
+	}{
+		{"fixtures/envdir/simple", []string{"BAR=baz", "FOO=bar"}, false},
+		{"fixtures/envdir/firstline", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/junk", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/trim", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/null", []string{"FOO=ba\nr"}, false},
+		{"fixtures/envdir/empty", []string{}, false},
+		{"fixtures/envdir/whitespace", []string{}, true},
+	}
+
+	for _, f := range fixtures {
+		vars, err := ParseEnvDir(f.dir)
+		if f.failure {
+			if err == nil {
+				t.Fatal("Expected a failure, but succeeded instead.")
+			}
+		} else {
+			if err != nil {
+				t.Fatal(err)
```

Comment:
can we get the 'dir' (or something) printed so we know which test we were on when it failed?  Just in case 'err' doesn't contain anything to tell us.